### ISSUE TITLE
[IOS] 5.4" inch screen support

### ISSUE THREAD

**Issue by EliasDeuss** [flutter/flutter#62437]
Looks like Apple will be launching a 5.4" inch iPhone this year. 

In iOS 14 Beta 3+. In Display & Brightness there is a Display Zoom menu where you can chose a zoomed in mode. Which scales up the screen (see attached images). It looks like flutter assumes its running on a iPhone 8 or lower.

![IMG_0481](https://user-images.githubusercontent.com/13773566/88716094-65a1c980-d0d3-11ea-8446-88ccae6f3c76.png)

App running Flutter 1.20.0.7.2-Pre:
![IMG_0718](https://user-images.githubusercontent.com/13773566/91243582-2af57600-e6ff-11ea-89a0-cf68a67c6670.PNG)

![IMG_0719](https://user-images.githubusercontent.com/13773566/91243592-2df06680-e6ff-11ea-8d9e-603c136b7b42.PNG)

Native Apple App:

![IMG_0482](https://user-images.githubusercontent.com/13773566/88716403-dfd24e00-d0d3-11ea-9860-e63c729faedf.png)

![IMG_0483](https://user-images.githubusercontent.com/13773566/88716415-e4970200-d0d3-11ea-81a2-41acd781ae16.png)

```
[✓] Flutter (Channel stable, 1.20.2, on macOS 11.0 20A5354i, locale en-US)
    • Flutter version 1.20.2 at /Users/elias/flutter
    • Framework revision bbfbf1770c (12 days ago), 2020-08-13 08:33:09 -0700
    • Engine revision 9d5b21729f
    • Dart version 2.9.1

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/elias/Library/Android/sdk
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.6)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.6, Build version 11E708
    • CocoaPods version 1.9.1

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 48.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.48.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.13.2

```

**Comment by TahaTesser** [comment]
Hi @EliasDeuss 
I can see that zoom option in iOS 13.6 too, not sure it's worth  creating this issue before device is even is announced/released 
Do you have any issues when flutter in zoomed mode or is this just feature request for 5.4in support? 

if you've an issue with zoom, Can you please provide your `flutter doctor -v` your `flutter run --verbose` and a complete reproducible minimal code sample
Thank you

**Comment by EliasDeuss** [comment]
> Hi @EliasDeuss 
> 
> I can see that zoom option in iOS 13.6 too, not sure it's worth  creating this issue before device is even is announced/released 
> 
> Do you have any issues when flutter in zoomed mode or is this just feature request for 5.4in support? 
> 
> 
> 
> if you've an issue with zoom, Can you please provide your `flutter doctor -v` your `flutter run --verbose` and a complete reproducible minimal code sample
> 
> Thank you

@TahaTesser Well it’s just the flutter app doesn’t recognize that it’s zoomed in so it doesn’t use the top or bottom of the screen. So you have empty space (compared to apples default apps).

**Comment by TahaTesser** [comment]
Hi @EliasDeuss 
I can there is difference when using zoom mode in Flutter
### Left (normal) & right (zoomed)

<img width="830" alt="Screenshot 2020-07-29 at 5 19 01 PM" src="https://user-images.githubusercontent.com/48603081/88796679-aaa32a00-d1bf-11ea-821e-b0bebd9e1897.png">

Do you see different behaviour? 
Can you please provide your `flutter doctor -v` your `flutter run --verbose` and a complete reproducible minimal code sample
Thank you

**Comment by TahaTesser** [comment]
Without additional information we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

**Comment by EliasDeuss** [comment]
@TahaTesser I did more research and it looks like it was added in iOS 14 Beta 3. So if you have your phone on zoomed mode in that beta or above it will behave like that.

**Comment by EliasDeuss** [comment]
@jmagman You might want to add this to #61047 . This looks like to be a iOS 14 Beta 3 addition, also here is an article talking about it. [Article Link](https://9to5mac.com/2020/07/22/ios-14-beta-gives-us-a-glimpse-of-the-system-interface-on-the-5-4-inch-iphone-12/)

**Comment by EliasDeuss** [comment]
@jmagman I also just updated the screenshots to show light mode. You can see the issue a lot better with the black bars.

**Comment by jmagman** [comment]
I tried the zoom on the gallery app on an iPad Pro and it looks pretty good.  Flutter handles all kinds of screen sizes responsively (particularly on Android), so I wouldn't expect a 5.4" screen to cause any issues.
Standard:
![Screen Shot 2020-08-25 at 6 45 42 PM](https://user-images.githubusercontent.com/682784/91245825-46628000-e703-11ea-9227-5d1247ecfc48.png)
(I don't know why the screenshot is cropped, but it looks good on the device.)
Zoomed:
![Screen Shot 2020-08-25 at 6 44 30 PM](https://user-images.githubusercontent.com/682784/91245875-62662180-e703-11ea-86a0-557e10159f17.png)

Could you give us a sample app that reproduces this?

**Comment by EliasDeuss** [comment]
@jmagman Here is the flutter gallery app.

```
[✓] Flutter (Channel stable, 1.20.2, on macOS 11.0 20A5354i, locale en-US)
    • Flutter version 1.20.2 at /Users/elias/flutter
    • Framework revision bbfbf1770c (12 days ago), 2020-08-13 08:33:09 -0700
    • Engine revision 9d5b21729f
    • Dart version 2.9.1
```

![IMG_0722](https://user-images.githubusercontent.com/13773566/91347283-e87f7800-e796-11ea-9078-7d0efec08df6.PNG)
![IMG_0723](https://user-images.githubusercontent.com/13773566/91347287-eb7a6880-e796-11ea-9cd5-092fad0589c6.PNG)

**Comment by jmagman** [comment]
@EliasDeuss That looks pretty good to me.  We can hold this open until someone can test with a real 5.4" device when it ships.  I suspect it will scale fine based on all the other phone screen sizes Flutter supports.

**Comment by jmagman** [comment]
5.4" iPhone 12 mini simulator gallery looks good.
![simulator_screenshot_E8B1BDA1-EDA9-479A-957F-8D9311DE7BE4](https://user-images.githubusercontent.com/682784/97038085-1e36a800-151f-11eb-865b-fb7f9a559ba5.png)

Closing.  If you see bugs related to the new size, please file new issues and completely fill out the template.

**Comment by vanlooverenkoen** [comment]
Is there a way to disable this? One of our clients wants this zoomed behaviour disabled for their app.

**Comment by jmagman** [comment]
@vanlooverenkoen Please file a new issue and completely fill out the template so the frameworks team can take a look.
